src/main/java/com/vw/ipppdiffer/controller/DifferController.java

```diff
@@ -17,11 +21,19 @@
 
     @Override
     public DifferResponse startDiffer(MultipartFile firstFile, MultipartFile secondFile) {
-        return service.getDiffer(firstFile, secondFile);
+        try {
+            return service.getDiffer(firstFile, secondFile);
+        } catch (JAXBException | IOException | IllegalAccessException e) {
+            throw new UnknownParseException("Differ failed: Failed to parse file!");
+        }
     }
 
     @Override
     public Element getTree(MultipartFile file) {
-        return service.getTreeStructure(file);
+        try {
+            return service.getTreeStructure(file);
+        } catch (JAXBException | IOException | IllegalAccessException e) {
+            throw new UnknownParseException("Failed to parse file!");
```

Comment:
Throw the UnknownParseException from the service and catch it here.

Reply:
I don't think that I need to catch it in this case since the global @ControllerAdvice will  handle the custom exception regardless from where is trow.

Reply:
The idea is that the Controller don't need to know about the lower level exceptions, it needs to catch the exception what the service builds. 
The general rule is that catch the exception in lower layers and throw layer specific exceptions to the higher layers.

Reply:
But if the controller catch the exception and for example logs it than the @ControllerAdvice lose it's meaning . In this case the exception is throw from the service and is handled by ControllerExceptionHandler and will return to the client error code 400 and the proper message.